File: ksl/process/test/system/hello_world_mpi.c
#include <stdio.h>
#include <mpi.h>

int main (int argc, char *argv[])
{
  int rank, size;
  MPI_Init (&argc, &argv);
  MPI_Comm_rank (MPI_COMM_WORLD, &rank);
  MPI_Comm_size (MPI_COMM_WORLD, &size);
  if (rank==0)
    printf( "%d.%d: hello world\n", size, rank);
  MPI_Finalize();
  return 0;
} 

File: ksl/process/test/system/hello_world_mpi.f90
program hello_world_mpi
  use mpi
  integer size, rank, ierr
  call MPI_INIT(ierr)
  call MPI_COMM_SIZE(MPI_COMM_WORLD, size, ierr)
  call MPI_COMM_RANK(MPI_COMM_WORLD, rank, ierr)
  if (rank .eq. 0) write(*,'(I6,A,I1,A)') size, '.', rank, ': hello world'
  call MPI_FINALIZE(ierr)
end program hello_world_mpi
